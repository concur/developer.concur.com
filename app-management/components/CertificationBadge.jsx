import React, { PropTypes } from 'react';

const CertificationBadge = ({ certified }) => (
  certified ? (
    <small className="certification-badge certified">
      <i className="fa fa-check-circle" aria-hidden="true" /> CERTIFIED
    </small>
  ) : (
    <small className="certification-badge not-certified">
      <i className="fa fa-times-circle" aria-hidden="true" /> NOT CERTIFIED
    </small>
  )
);

CertificationBadge.propTypes = {
  certified: PropTypes.bool,
};

export default CertificationBadge;
