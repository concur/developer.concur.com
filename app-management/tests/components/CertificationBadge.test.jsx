import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CertificationBadge from '../../components/CertificationBadge';

describe('<CertificationBadge />', () => {
  it('renders "CERTIFIED" if certified', () => {
    const badge = shallow(<CertificationBadge certified={true} />);

    expect(shallowToJson(badge)).toMatchSnapshot();
  });

  it('renders "NOT CERTIFIED" if not certified', () => {
    const badge = shallow(<CertificationBadge certified={false} />);

    expect(shallowToJson(badge)).toMatchSnapshot();
  });
});
