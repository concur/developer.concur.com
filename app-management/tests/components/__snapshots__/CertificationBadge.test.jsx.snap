exports[`<CertificationBadge /> renders "CERTIFIED" if certified 1`] = `
<small
  className="certification-badge certified">
  <i
    aria-hidden="true"
    className="fa fa-check-circle" />
   CERTIFIED
</small>
`;

exports[`<CertificationBadge /> renders "NOT CERTIFIED" if not certified 1`] = `
<small
  className="certification-badge not-certified">
  <i
    aria-hidden="true"
    className="fa fa-times-circle" />
   NOT CERTIFIED
</small>
`;
